src/main/java/org/opensearch/knn/index/engine/faiss/FaissHNSWMethod.java

```diff
@@ -154,34 +160,55 @@ protected Function<TrainingConfigValidationInput, TrainingConfigValidationOutput
+        builder.spaceType(getStringFromMap(parameters, SPACE_TYPE));
 
-        MethodComponentContext methodComponentContext = knnMethodContext.getMethodComponentContext();
-        Map<String, Object> methodParams = methodComponentContext.getParameters();
-        builder.efConstruction(getIntegerFromMap(methodParams, METHOD_PARAMETER_EF_CONSTRUCTION));
-        builder.efSearch(getIntegerFromMap(methodParams, METHOD_PARAMETER_EF_SEARCH));
-        builder.m(getIntegerFromMap(methodParams, METHOD_PARAMETER_M));
+        Map<String, Object> innerParameters = (Map<String, Object>) parameters.get(PARAMETERS);
+        builder.efConstruction(getIntegerFromMap(innerParameters, METHOD_PARAMETER_EF_CONSTRUCTION));
+        builder.efSearch(getIntegerFromMap(innerParameters, METHOD_PARAMETER_EF_SEARCH));
+        builder.m(getIntegerFromMap(innerParameters, METHOD_PARAMETER_M));
         return builder.build();
     }
 
     /**
-     * @param methodParameters      Map of method parameters including encoder information
-     * @param vectorDataType        {@link VectorDataType}
-     * @return                      true if the method parameters + vector data type combination is supported for remote index build
+     * @param parameters Map of method parameters including encoder information
+     * Example JSON structure:
+     * {
+     *   "index_description": "HNSW12,Flat",
+     *   "spaceType": "innerproduct",
+     *   "name": "hnsw",
+     *   "data_type": "float",
+     *   "parameters": {
+     *     "ef_search": 24,
+     *     "ef_construction": 28,
+     *     "m": 12,
+     *     "encoder": {
+     *       "name": "flat",
+     *       "parameters": {}
+     *     }
+     *   }
+     * }
+     * @return true if the method parameters + vector data type combination is supported for remote index build
      */
-    static boolean supportsRemoteIndexBuild(Map<String, Object> methodParameters, VectorDataType vectorDataType) {
-        if (SUPPORTED_REMOTE_INDEX_DATA_TYPES.contains(vectorDataType)) {
-            String encoderName = ((MethodComponentContext) methodParameters.get(METHOD_ENCODER_PARAMETER)).getName();
-            return ENCODER_FLAT.equals(encoderName);
+    @SuppressWarnings("unchecked")
+    static boolean supportsRemoteIndexBuild(Map<String, Object> parameters) {
+        try {
+            Map<String, Object> innerMap = (Map<String, Object>) parameters.get(PARAMETERS);
+            Map<String, Object> encoderMap = (Map<String, Object>) innerMap.get(METHOD_ENCODER_PARAMETER);
+            String dataType = getStringFromMap(parameters, VECTOR_DATA_TYPE_FIELD);
+            String encoder = getStringFromMap(encoderMap, NAME);
+            return SUPPORTED_REMOTE_INDEX_DATA_TYPES.contains(VectorDataType.get(dataType)) && ENCODER_FLAT.equals(encoder);
+        } catch (IllegalArgumentException e) {
+            log.error("Unrecognized indexing parameters in KNNLibraryIndexingContext");
```

Comment:
```suggestion
            log.error("Unrecognized indexing parameters in KNNLibraryIndexingContext ", e);
```

Lets also log what was the indexing parameters